### PR TITLE
fix(forms/TDC-1885): focus the first error after submit

### PR DESCRIPTION
**Pull request by jare-talend**
**What is the problem this PR is trying to solve?**
when a wide form is invalid and the user submits the form we can't see why we can't valid a form.

**What is the chosen solution to this problem?**
We focus the first element in error on the form

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [X] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: output/forms.eslint.txt [hunk 1]
```diff
@@ -5,5 +5,8 @@ The react/require-extension rule is deprecated. Please use the import/extensions
 /home/travis/build/Talend/ui/packages/forms/src/fields/CollapsibleFieldset.js
   82:8  error  Visible, non-interactive elements should not have mouse or keyboard event listeners  jsx-a11y/no-static-element-interactions
 
-✖ 1 problem (1 error, 0 warnings)
+/home/travis/build/Talend/ui/packages/forms/src/UIForm/UIForm.component.test.js
+  6:18  error  Absolute imports should come before relative imports  import/imports-first
```
Comment (romainseb) [inline output/forms.eslint.txt]: Could you fix this lint ?

Comment (jare-talend) [reply output/forms.eslint.txt]: done

File: packages/forms/src/UIForm/UIForm.component.js [hunk 2]
```diff
@@ -231,19 +236,32 @@ export class UIFormComponent extends React.Component {
 				return accu;
 			}, {});
 
-		this.props.setErrors(event, errors);
-
 		const isValid = !Object.keys(errors).length;
+		this.props.setErrors(event, errors, isValid ? noop : this.focusFirstError);
```
Comment (jsomsanith) [inline packages/forms/src/UIForm/UIForm.component.js]: Please remove the noop. Pass undefined

File: packages/forms/src/UIForm/UIForm.component.test.js [hunk 3]
```diff
@@ -252,6 +253,11 @@ describe('UIForm component', () => {
 	});
 
 	describe('#onSubmit', () => {
+		const validProperties = {
```
Comment (jsomsanith) [inline packages/forms/src/UIForm/UIForm.component.test.js]: This seems to be used in 1 test, please move that back in the test

File: packages/forms/src/UIForm/UIForm.component.test.js [hunk 4]
```diff
@@ -273,9 +279,13 @@ describe('UIForm component', () => {
...
 			// then
-			expect(props.setErrors).toBeCalledWith(submitEvent, {
-				firstname: 'Missing required field',
-			});
+			expect(props.setErrors).toBeCalledWith(
+				submitEvent,
+				{
+					firstname: 'Missing required field',
+				},
+				wrapper.instance().focusFirstError,
```
Comment (jsomsanith) [inline packages/forms/src/UIForm/UIForm.component.test.js]: Replace that to `expect.anything()`.
This is not the goal of this test, and this 3rd arg is tested in a specific test

File: packages/forms/src/UIForm/UIForm.component.test.js [hunk 5]
```diff
@@ -295,11 +305,15 @@ describe('UIForm component', () => {
...
-				check: 'error added via a trigger',
-			});
+			expect(props.setErrors).toBeCalledWith(
+				submitEvent,
+				{
+					firstname: 'Missing required field',
+					lastname: 'String is too short (6 chars), minimum 10',
+					check: 'error added via a trigger',
+				},
+				wrapper.instance().focusFirstError,
```
Comment (jsomsanith) [inline packages/forms/src/UIForm/UIForm.component.test.js]: Same here

File: packages/forms/src/UIForm/UIForm.component.test.js [hunk 6]
```diff
@@ -314,9 +328,13 @@ describe('UIForm component', () => {
...
 			// then
-			expect(props.setErrors).toBeCalledWith(submitEvent, {
-				firstname: 'is required',
-			});
+			expect(props.setErrors).toBeCalledWith(
+				submitEvent,
+				{
+					firstname: 'is required',
+				},
+				wrapper.instance().focusFirstError,
```
Comment (jsomsanith) [inline packages/forms/src/UIForm/UIForm.component.test.js]: same here

File: packages/forms/src/UIForm/UIForm.component.test.js [hunk 7]
```diff
@@ -332,18 +350,39 @@ describe('UIForm component', () => {
...
+
+	describe('#focusFirstError', () => {
+		it('should focus the first element', () => {
+			const focus = jest.fn();
+			const querySelector = jest.fn(() => ({
+				focus,
+			}));
+			const wrapper = shallow(<UIFormComponent {...data} {...props} />);
+			wrapper.instance().formRef = { querySelector };
+			wrapper.instance().focusFirstError();
```
Comment (jsomsanith) [inline packages/forms/src/UIForm/UIForm.component.test.js]: Instead of getting an internal function, can you
1- pass a mock as setErrors
2- mount it
3- submit the form with errors
4- get the 3rd argument of setErrors
5- call it
6- check the activeElement

File: packages/forms/src/UIForm/UIForm.component.test.js [hunk 8]
```diff
@@ -332,18 +350,39 @@ describe('UIForm component', () => {
...
+
+			expect(querySelector).toHaveBeenCalledWith(INVALID_FIELD_SELECTOR);
+			expect(focus).toHaveBeenCalled();
+		});
+	});
+
+	describe('#setFormRef', () => {
```
Comment (jsomsanith) [inline packages/forms/src/UIForm/UIForm.component.test.js]: Remove this test, it's about internal implem, it's indirectly tested with the focus

File: packages/forms/src/UIForm/UIForm.component.js (hunk 2 above)
Comment (jare-talend) [reply packages/forms/src/UIForm/UIForm.component.js]: done

File: packages/forms/src/UIForm/UIForm.component.test.js (hunk 3 above)
Comment (jare-talend) [reply packages/forms/src/UIForm/UIForm.component.test.js]: done

File: packages/forms/src/UIForm/UIForm.component.test.js (hunk 4 above)
Comment (jare-talend) [reply packages/forms/src/UIForm/UIForm.component.test.js]: done

File: packages/forms/src/UIForm/UIForm.component.test.js (hunk 5 above)
Comment (jare-talend) [reply packages/forms/src/UIForm/UIForm.component.test.js]: done

File: packages/forms/src/UIForm/UIForm.component.test.js (hunk 6 above)
Comment (jare-talend) [reply packages/forms/src/UIForm/UIForm.component.test.js]: done

File: packages/forms/src/UIForm/UIForm.component.test.js (hunk 7 above)
Comment (jare-talend) [reply packages/forms/src/UIForm/UIForm.component.test.js]: done

File: packages/forms/src/UIForm/UIForm.component.test.js (hunk 8 above)
Comment (jare-talend) [reply packages/forms/src/UIForm/UIForm.component.test.js]: done

File: packages/forms/src/UIForm/UIForm.component.js [hunk 9]
```diff
@@ -18,6 +18,10 @@ import { I18N_DOMAIN_FORMS } from '../constants';
 import '../translate';
 import theme from './UIForm.scss';
 
+export const INVALID_ATTRIBUTE = 'aria-invalid';
+
+const INVALID_SELECTOR = `[${INVALID_ATTRIBUTE}="true"]`;
```
Comment (jsomsanith) [inline packages/forms/src/UIForm/UIForm.component.js]: This feels like `const selector = 'div'` and use `selector` everywhere instead of `div`.
You can remove those 2 vars and inline the string in the only function that use it, and in the test. It's not something that will change.

File: packages/forms/src/UIForm/UIForm.component.test.js [hunk 10]
```diff
@@ -331,19 +343,37 @@ describe('UIForm component', () => {
 		});
 
 		it('should call submit callback when form is valid', () => {
-			// given
 			const validProperties = {
 				...data.properties,
 				lastname: 'This has at least 10 characters',
 				firstname: 'This is required',
 			};
+			// given
```
Comment (jsomsanith) [inline packages/forms/src/UIForm/UIForm.component.test.js]: wat
The const just above is part of the `given`, why did you move this comment ?
